core/src/test/java/org/mskcc/oncotree/utils/TumorTypesUtilTest.java

```diff
@@ -0,0 +1,866 @@
+    }
+
+    private void setupForTest() throws Exception {
+        setupMockRepository();
+        setupTumorTypesUtil();
+        setupExpectedTumorTypeMap();
+    }
+
+    public String makeMismatchMessage(String oncoTreeCode, String fieldName, String gotValue, String expectedValue) {
+        return oncoTreeCode + " : mismatch in " + fieldName + " (got:" + gotValue + ") (expected:" + expectedValue + ")\n";
+    }
+
+    public Set<TumorType> simulateTumorTypeApiResponse() throws Exception {
+        Map<String, TumorType> tumorTypeMap = tumorTypesUtil.getTumorTypesByVersionFromRaw(mockVersion);
+        Set<TumorType> tumorTypeSet = tumorTypesUtil.flattenTumorTypes(tumorTypeMap, null);
+        return tumorTypeSet;
+    }
+
+    private boolean testValuesMatch(String value1, String value2) {
+        if (value1 == null || value1.trim().length() == 0) {
+            return value2 == null || value2.trim().length() == 0;
+        }
+        return value1.equals(value2);
+    }
+
+    @Test
+    public void testFullOncoTreeCodeListByCode() throws Exception {
+        setupForTest();
+        Set<TumorType> tumorTypesAsPreparedForTumorTypesEndpoint = simulateTumorTypeApiResponse();
+        HashMap<String,TumorType> returnedTumorTypeMap = new HashMap<>(tumorTypesAsPreparedForTumorTypesEndpoint.size());
+        for (TumorType tumorType : tumorTypesAsPreparedForTumorTypesEndpoint) {
+            returnedTumorTypeMap.put(tumorType.getCode(), tumorType);
+        }
+        int failureCount = 0;
+        StringBuilder failureReport  = new StringBuilder();
+        for (String oncoTreeCode : expectedTumorTypeMap.keySet()) {
+            TumorType expectedTumorType = expectedTumorTypeMap.get(oncoTreeCode);
+            TumorType returnedTumorType = returnedTumorTypeMap.get(oncoTreeCode);
+            if (expectedTumorType == null || returnedTumorType == null) {
+                if (expectedTumorType == null) {
+                    failureReport.append(oncoTreeCode + " : no values in expected tumor type hash (internal error)\n");
+                }
+                if (returnedTumorType == null) {
+                    failureReport.append(oncoTreeCode + " : tumor type expected but not returned from tumorTypesUtil\n");
+                }
+            } else {
+                if (!expectedTumorType.getCode().equals(returnedTumorType.getCode())) {
+                    failureReport.append(makeMismatchMessage(oncoTreeCode, "code", returnedTumorType.getCode(), expectedTumorType.getCode()));
+                    failureCount = failureCount + 1;
+                }
+                if (!expectedTumorType.getName().equals(returnedTumorType.getName())) {
+                    failureReport.append(makeMismatchMessage(oncoTreeCode, "name", returnedTumorType.getName(), expectedTumorType.getName()));
+                    failureCount = failureCount + 1;
+                }
+                if (!expectedTumorType.getColor().equals(returnedTumorType.getColor())) {
```

Comment:
I changed getColor() in OncoTreeNode.java to return "blue" and successfully failed the tests:

Failed tests: 
  TumorTypesUtilTest.testFullOncoTreeCodeListByCode:212 644 failed test conditions. Details follow... HGSOS : mismatch in color (got:blue) (expected:White)
TET : mismatch in color (got:blue) (expected:Purple)
BLSC : mismatch in color (got:blue) (expected:Yellow)
RAS : mismatch in color (got:blue) (expected:LightYellow)
SCT : mismatch in color (got:blue) (expected:LightBlue)
HGNET : mismatch in color (got:blue) (expected:Gray)